src/app/routes/assets/asset-enrolment-list/asset-enrolment-list.component.ts

```diff
@@ -51,6 +51,18 @@ export class AssetEnrolmentListComponent implements OnInit, OnDestroy {
   }
 
   back() {
-    this.location.back();
+    this.urlService.previous.pipe(
+      takeUntil(this.subscription$)
+    ).subscribe(url => {
+      this.navigateBack(url);
+    });
+  }
+
+  navigateBack(url: string): void {
+    if (url.includes('returnUrl')) {
```

Comment:
I think a comment could be valuable here, as I think that it is not immediately obvious why this logic is in place
```suggestion
    // 'returnUrl' is taken as an indicator back() would trigger an loop back to asset-enrolment
    if (url.includes('returnUrl')) {
```